src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationHandler.java

```diff
@@ -45,10 +59,75 @@ public List<CaseEvent> handledEvents() {
     }
 
     private CallbackResponse sendNotifications(CallbackParams callbackParams) {
-        final String taskId = callbackParams.getCaseData().getBusinessProcess().getActivityId();
-        log.info("Sending notifications for taskId: {}", taskId);
+        final CaseData caseData = callbackParams.getCaseData();
+        final String taskId = caseData.getBusinessProcess().getActivityId();
         final Notifier notifier = notifierFactory.getNotifier(taskId);
-        notifier.notifyParties(callbackParams.getCaseData(), NOTIFY_EVENT.toString(), taskId);
-        return AboutToStartOrSubmitCallbackResponse.builder().build();
+        final String summary = notifier.notifyParties(caseData, NOTIFY_EVENT.toString(), taskId);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseData.toBuilder().notificationSummary(summary).build().toMap(objectMapper))
+            .build();
+    }
+
+    private CallbackResponse recordNotifications(CallbackParams callbackParams) {
+        final CaseData caseData = callbackParams.getCaseData();
```

Comment:
as far as I know - submitted callback do not commit data change? might want to put this in about to submit